### DISCUSSION TITLE
Private comments in public repositories

### DISCUSSION THREAD

**Discussion by zbjornson** [community/community#15806] [Projects and Issues]
Sometimes we want to have private discussions in a public repository's issues. For example, maybe we need to discuss a security matter or discuss some private server code in the context of a public client library. In those cases, it would be nice to post comments that are only visible to maintainers or a selected Team.

**Comment by hzongaro** [comment]
Another situation where one might want to use a private discussion would be to point out in private some inappropriate behaviour to help guide rather than publicly shame an individual.

**Comment by Scoup** [comment]
Any update about this issue?

**Comment by DSDmark** [comment]
It may be more great to separate the communication channels, like as private chats or dedicated collaboration tools, for sensitive or confidential discussions.

**Comment by romw314** [comment]
Try [Chatzy](https://us28.chatzy.com).

**Comment by hzongaro** [comment]
If I understand, the root of the problem is that there doesn't seem to be a way to contact a Github user or group of users directly if they haven't made their e-mail addresses public.

**Comment by ryahconstantino** [comment]
Testing if my comment is public or not

**Comment by Dave247** [comment]
My thought is taken to the [Google Issue Tracker](https://issuetracker.google.com/), where file attachments can be made private to various levels of Google staff when it comes to users logs and dumps.

This could be used with an organisation's team, where you can choose to post a reply that is only visible to yourself and that team. Note that you should not have to be part of that team yourself to select it, as your own replies should always be visible to you.

**Reply by roke-julian-lockhart** [reply]
@Dave247, GitLab offers this (and is actually usable outside of Google). [^2] It even permits private issues in public repositories. [^1]

[^1]: [`discussion.fedoraproject.org/t/168745/23`](https://discussion.fedoraproject.org/t/workaround-for-missing-private-issues-in-public-repositories/168745/23)

[^2]: [`webapps.stackexchange.com/questions/181841/is-google-buganizer-self-hostable#comment171674_181841`](https://webapps.stackexchange.com/questions/181841/is-google-buganizer-self-hostable#comment171674_181841)

**Comment by kring** [comment]
Would love to see this!

We use GitHub for both our open- and closed-source development. It's often useful to be able to add a comment to an issue in a public repo that should not be shared beyond the maintainers. For example, we might add a note that a particular feature in an open-source product is needed for an as-yet-unannounced closed-source one.

We have other channels we can use for private conversation, of course, but associating such comments with the issues themselves makes them much easier to discover.

The workaround is issue linking. If we mention a public repo issue from a private repo one, GitHub helpfully creates a bi-directional link. This is fantastic, but we often want to provide more context (not just a link).

**Comment by rbetts** [comment]
Similar to kring's comment, another currently unmet use case is to allow CI bots to publish commercially proprietary CI results (like competitive benchmark numbers) to PRs in public repositories.

**Comment by psjamesh** [comment]
+1 for this. We're currently using JIRA to manage progress of an issue internally because we cannot comment on the issue without these being publicly visible.

**Comment by rathboma** [comment]
Echoing the use cases here plus another one - attaching a customer-provided screenshare or screenshot to a bug report, but not making it public to everyone on the internet.

I need to do this frequently.

**Comment by flavienbwk** [comment]
Although I supported this feature some times ago, with all these recent security issues at GitHub, maybe it's better to not have private comments at the same place in the end...

**Comment by ncreated** [comment]
Another pattern that seems likely to become common: LLM-based code review agents running on public repositories with access to internal knowledge. Today there's no way to keep the review output private to maintainers — so any internal signal the agent drew on ends up visible to everyone.
